File: server-scala-play-server/app/api/CargoTypeApiController.scala
package api

import org.openapitools.OpenApiExceptions
import javax.inject.{Inject, Singleton}
import play.api.libs.json._
import play.api.mvc._
import model.BigDecimal
import model.CargoType

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
@Singleton
class CargoTypeApiController @Inject()(cc: ControllerComponents, api: CargoTypeApi) extends AbstractController(cc) {
  /**
    * POST /api/:version/cargo/type
    */
  def createCargoType(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): CargoType = {
      val body = request.body.asJson.map(_.as[CargoType])
      api.createCargoType(version, body)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * DELETE /api/:version/cargo/type/:cargoTypeId
    * @param cargoTypeId the ID of the cargo type
    */
  def deleteCargoType(version: BigDecimal, cargoTypeId: Long): Action[AnyContent] = Action { request =>
    def executeApi(): Unit = {
      api.deleteCargoType(version, cargoTypeId)
    }

    executeApi()
    Ok
  }

  /**
    * GET /api/:version/cargo/type/:cargoTypeId
    * @param cargoTypeId the cargo type ID
    */
  def getCargoType(version: BigDecimal, cargoTypeId: Long): Action[AnyContent] = Action { request =>
    def executeApi(): CargoType = {
      api.getCargoType(version, cargoTypeId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * GET /api/:version/cargo/type?retailerId=[value]&hubId=[value]&sortField=[value]&descending=[value]&start=[value]&limit=[value]&activeOnly=[value]
    */
  def searchCargoTypes(version: BigDecimal): Action[AnyContent] = Action { request =>
    def executeApi(): List[CargoType] = {
      val retailerId = request.getQueryString("retailerId")
        .map(value => value.toLong)
        
      val hubId = request.getQueryString("hubId")
        .map(value => value.toLong)
        
      val sortField = request.getQueryString("sortField")
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("sortField", "query string")
        }
        
      val descending = request.getQueryString("descending")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("descending", "query string")
        }
        
      val start = request.getQueryString("start")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("start", "query string")
        }
        
      val limit = request.getQueryString("limit")
        .map(value => value.toInt)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("limit", "query string")
        }
        
      val activeOnly = request.getQueryString("activeOnly")
        .map(value => value.toBoolean)
        .getOrElse {
          throw new OpenApiExceptions.MissingRequiredParameterException("activeOnly", "query string")
        }
        
      api.searchCargoTypes(version, sortField, descending, start, limit, activeOnly, retailerId, hubId)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  /**
    * PUT /api/:version/cargo/type/:cargoTypeId
    * @param cargoTypeId the ID of the cargo type
    */
  def updateCargoType(version: BigDecimal, cargoTypeId: Long): Action[AnyContent] = Action { request =>
    def executeApi(): CargoType = {
      val body = request.body.asJson.map(_.as[CargoType])
      api.updateCargoType(version, cargoTypeId, body)
    }

    val result = executeApi()
    val json = Json.toJson(result)
    Ok(json)
  }

  private def splitCollectionParam(paramValues: String, collectionFormat: String): List[String] = {
    val splitBy =
      collectionFormat match {
        case "csv" => ",+"
        case "tsv" => "\t+"
        case "ssv" => " +"
        case "pipes" => "|+"
      }

    paramValues.split(splitBy).toList
  }
}

File: server-java-play-framework/app/controllers/CargoTypeApiController.java
package controllers;

import java.math.BigDecimal;
import apimodels.CargoType;

import com.typesafe.config.Config;
import play.mvc.Controller;
import play.mvc.Result;
import play.mvc.Http;
import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.JsonNode;
import com.google.inject.Inject;
import java.io.File;
import play.libs.Files.TemporaryFile;
import openapitools.OpenAPIUtils;
import com.fasterxml.jackson.core.type.TypeReference;

import javax.validation.constraints.*;
import javax.validation.Valid;
import com.typesafe.config.Config;

import openapitools.OpenAPIUtils.ApiAction;

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPlayFrameworkCodegen", date = "2026-02-09T20:49:20.380387445Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
public class CargoTypeApiController extends Controller {
    private final CargoTypeApiControllerImpInterface imp;
    private final ObjectMapper mapper;
    private final Config configuration;

    @Inject
    private CargoTypeApiController(Config configuration, CargoTypeApiControllerImpInterface imp) {
        this.imp = imp;
        mapper = new ObjectMapper();
        this.configuration = configuration;
    }

    @ApiAction
    public Result createCargoType(Http.Request request, BigDecimal version) throws Exception {
        JsonNode nodebody = request.body().asJson();
        CargoType body;
        if (nodebody != null) {
            body = mapper.readValue(nodebody.toString(), CargoType.class);
            if (configuration.getBoolean("useInputBeanValidation")) {
                OpenAPIUtils.validate(body);
            }
        } else {
            body = null;
        }
        return imp.createCargoTypeHttp(request, version, body);
    }

    @ApiAction
    public Result deleteCargoType(Http.Request request, BigDecimal version,Long cargoTypeId) throws Exception {
        return imp.deleteCargoTypeHttp(request, version, cargoTypeId);
    }

    @ApiAction
    public Result getCargoType(Http.Request request, BigDecimal version,Long cargoTypeId) throws Exception {
        return imp.getCargoTypeHttp(request, version, cargoTypeId);
    }

    @ApiAction
    public Result searchCargoTypes(Http.Request request, BigDecimal version) throws Exception {
        String valueretailerId = request.getQueryString("retailerId");
        Long retailerId;
        if (valueretailerId != null) {
            retailerId = Long.parseLong(valueretailerId);
        } else {
            retailerId = null;
        }
        String valuehubId = request.getQueryString("hubId");
        Long hubId;
        if (valuehubId != null) {
            hubId = Long.parseLong(valuehubId);
        } else {
            hubId = null;
        }
        String valuesortField = request.getQueryString("sortField");
        String sortField;
        if (valuesortField != null) {
            sortField = valuesortField;
        } else {
            throw new IllegalArgumentException("'sortField' parameter is required");
        }
        String valuedescending = request.getQueryString("descending");
        Boolean descending;
        if (valuedescending != null) {
            descending = Boolean.valueOf(valuedescending);
        } else {
            throw new IllegalArgumentException("'descending' parameter is required");
        }
        String valuestart = request.getQueryString("start");
        Integer start;
        if (valuestart != null) {
            start = Integer.parseInt(valuestart);
        } else {
            throw new IllegalArgumentException("'start' parameter is required");
        }
        String valuelimit = request.getQueryString("limit");
        Integer limit;
        if (valuelimit != null) {
            limit = Integer.parseInt(valuelimit);
        } else {
            throw new IllegalArgumentException("'limit' parameter is required");
        }
        String valueactiveOnly = request.getQueryString("activeOnly");
        Boolean activeOnly;
        if (valueactiveOnly != null) {
            activeOnly = Boolean.valueOf(valueactiveOnly);
        } else {
            throw new IllegalArgumentException("'activeOnly' parameter is required");
        }
        return imp.searchCargoTypesHttp(request, version, sortField, descending, start, limit, activeOnly, retailerId, hubId);
    }

    @ApiAction
    public Result updateCargoType(Http.Request request, BigDecimal version,Long cargoTypeId) throws Exception {
        JsonNode nodebody = request.body().asJson();
        CargoType body;
        if (nodebody != null) {
            body = mapper.readValue(nodebody.toString(), CargoType.class);
            if (configuration.getBoolean("useInputBeanValidation")) {
                OpenAPIUtils.validate(body);
            }
        } else {
            body = null;
        }
        return imp.updateCargoTypeHttp(request, version, cargoTypeId, body);
    }

}
